### PR TITLE
feat: 🚀 Paths and PathValue

### PR DESCRIPTION
**Pull request by Beraliv**
**Summary**

- Implemented `Paths` – https://github.com/ts-essentials/ts-essentials/pull/303/commits/9d6faf9dbf2269de5eff00b0d80bd3f83b17368c
- Implemented `PathValue` – https://github.com/ts-essentials/ts-essentials/pull/303/commits/68287a4daf63b0052eab3ed257af6ad1b56d501e
- Removed TS 4.1 support because of excessive calculations for `PathValue` (https://github.com/ts-essentials/ts-essentials/actions/runs/8714151726/job/23903882806?pr=303) – https://github.com/ts-essentials/ts-essentials/pull/303/commits/ea3b1fcce90a1a382c5361c08b3ecab6fa518ca8, https://github.com/ts-essentials/ts-essentials/pull/303/commits/d983e727f4928859b9e0046aeb5a051a5e32c077

Related issue – https://github.com/ts-essentials/ts-essentials/issues/226

### CODE REVIEW

**Comment by welljsjs** [comment]
We'd be really interested in this feature! 

Personally got here from this SO post: https://stackoverflow.com/questions/58434389/typescript-deep-keyof-of-a-nested-object

**Comment by Beraliv** [comment]
@welljsjs @renataogarcia @VladSez I will need to bump TypeScript lower boundary with these 2 types, because TS 4.1 is quite upset with excessively deep and possibly infinite type instantiation

Is it okay for you?

**Comment by Beraliv** [comment]
cc @welljsjs @renataogarcia @GTOsss @VladSez I'm planning to merge it by the end of the month as it requires a breaking change (deprecation of support for TypeScript 4.1)

File: lib/path-value/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    : undefined
+  : never;
+
+type Path<Type> = Type extends `${infer Key}.${infer Rest}`
+  ? [Key, ...Path<Rest>]
+  : Type extends `${infer Key}`
+  ? [Key]
+  : [];
+
+export type PathValue<Type, StringPath> = GetWithArray<Type, Path<StringPath>>;
```
Comment (Beraliv) [inline lib/path-value/index.ts:33]: To optimise `PathValue` to avoid `Type instantiation is excessively deep and possibly infinite`

**Pull request merged**